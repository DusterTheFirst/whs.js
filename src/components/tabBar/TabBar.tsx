/*!
 * Copyright (C) 2018-2019  Zachary Kohnen (DusterTheFirst)
 */

import React from "react";
import { StyleSheet, View } from "react-native";
import TabBarLink from "./TabBarLink";

const styles = StyleSheet.create({
    bottomTabNav: {
        // backgroundColor: "#fff",
        borderTopColor: "#eee",
        borderTopWidth: 1,
        flexDirection: "row",
        height: 50,
        justifyContent: "space-around",
    },
});

export default function TabBar() {
    return (
        <View style={styles.bottomTabNav}>
            <TabBarLink icon="calendar" link="/calendar" name="Calendar"/>
            <TabBarLink icon="list" link="/today" name="Today"/>
            <TabBarLink icon="cog" link="/settings" name="Settings"/>
        </View>
    );
}