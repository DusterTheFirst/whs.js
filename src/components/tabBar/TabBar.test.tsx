/*!
 * Copyright (C) 2018-2019  Zachary Kohnen (DusterTheFirst)
 */

import React from "react";
import { render } from "react-native-testing-library";
import { NativeRouter } from "react-router-native";
import TabBar from "./TabBar";

it("Renders <TabBar/>", () => {
    let component = render(<NativeRouter><TabBar/></NativeRouter>);

    expect(component.toJSON()).toMatchSnapshot();
});