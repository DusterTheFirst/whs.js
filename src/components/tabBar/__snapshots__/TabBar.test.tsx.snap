// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders <TabBar/> 1`] = `
<View
  style={
    Object {
      "borderTopColor": "#eee",
      "borderTopWidth": 1,
      "flexDirection": "row",
      "height": 50,
      "justifyContent": "space-around",
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "margin": 5,
        "opacity": 1,
      }
    }
  >
    <View>
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#ccc",
              "fontSize": 22,
            },
            Object {
              "textAlign": "center",
            },
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#aaa",
              "fontSize": 13,
            },
            undefined,
          ]
        }
      >
        Calendar
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "margin": 5,
        "opacity": 1,
      }
    }
  >
    <View>
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#ccc",
              "fontSize": 22,
            },
            Object {
              "textAlign": "center",
            },
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#aaa",
              "fontSize": 13,
            },
            undefined,
          ]
        }
      >
        Today
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "margin": 5,
        "opacity": 1,
      }
    }
  >
    <View>
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#ccc",
              "fontSize": 22,
            },
            Object {
              "textAlign": "center",
            },
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#aaa",
              "fontSize": 13,
            },
            undefined,
          ]
        }
      >
        Settings
      </Text>
    </View>
  </View>
</View>
`;
